lighthouse-core/test/audits/image-size-responsive-test.js

```diff
@@ -0,0 +1,363 @@
+      naturalSize: [5, 5],
+      props: {
+        clientRect: {
+          top: 0,
+          bottom: 0,
+          left: 0,
+          right: 0,
+        },
+      },
+    });
+
+    testImage('is above the visible area', {
+      score: 1,
+      clientSize: [100, 100],
+      naturalSize: [5, 5],
+      props: {
+        clientRect: {
+          top: -201 - 100,
+          bottom: -201,
+          left: 0,
+          right: 100,
+        },
+      },
+    });
+
+    testImage('is almost above the visible area', {
+      score: 0,
+      clientSize: [100, 100],
+      naturalSize: [5, 5],
+      props: {
+        clientRect: {
+          top: -200 - 100,
+          bottom: -200,
+          left: 0,
+          right: 100,
+        },
+      },
+    });
+
+    testImage('is below the visible area', {
+      score: 1,
+      clientSize: [100, 100],
+      naturalSize: [5, 5],
+      props: {
+        clientRect: {
+          top: HEIGHT + 201,
+          bottom: HEIGHT + 201 + 100,
+          left: 0,
+          right: 100,
+        },
+      },
+    });
+
+    testImage('is almost below the visible area', {
+      score: 0,
+      clientSize: [100, 100],
+      naturalSize: [5, 5],
+      props: {
+        clientRect: {
+          top: HEIGHT + 200,
+          bottom: HEIGHT + 200 + 100,
+          left: 0,
+          right: 100,
+        },
+      },
+    });
+
+    testImage('is to the left of the visible area', {
+      score: 1,
+      clientSize: [100, 100],
+      naturalSize: [5, 5],
+      props: {
+        clientRect: {
+          top: 0,
+          bottom: 100,
+          left: -101 - 100,
+          right: -101,
+        },
+      },
+    });
+
+    testImage('is almost to the left of the visible area', {
+      score: 0,
+      clientSize: [100, 100],
+      naturalSize: [5, 5],
+      props: {
+        clientRect: {
+          top: 0,
+          bottom: 100,
+          left: -100 - 100,
+          right: -100,
+        },
+      },
+    });
+
+    testImage('is to the right of the visible area', {
+      score: 1,
+      clientSize: [100, 100],
+      naturalSize: [5, 5],
+      props: {
+        clientRect: {
+          top: 0,
+          bottom: 100,
+          left: WIDTH + 101,
+          right: WIDTH + 101 + 100,
+        },
+      },
+    });
+
+    testImage('is almost to the right of the visible area', {
+      score: 0,
+      clientSize: [100, 100],
+      naturalSize: [5, 5],
+      props: {
+        clientRect: {
+          top: 0,
+          bottom: 100,
+          left: WIDTH + 100,
+          right: WIDTH + 100 + 100,
+        },
+      },
+    });
+  });
+
+  describe('check size', () => {
+    describe('DPR = 1', () => {
+      testImage('is an icon with right size', {
+        score: 1,
+        clientSize: [64, 64],
+        naturalSize: [64, 64],
+      });
+
+      testImage('is an icon with an invalid size', {
+        score: 0,
+        clientSize: [64, 64],
+        naturalSize: [63, 63],
```

Comment:
oof, this seems a bit harsh, no?  do we really want 0 tolerance for them? I guess one might say these are the worst types of offenses though...

Reply:
It is, but it depends on the threshold we define. I wanted to keep it harsh for those sizes are those are the cases, in which I see the worst offenses, typically with iconography.

I refactored the code to have a `SMALL_IMAGE_FACTOR` and a `LARGE_IMAGE_FACTOR`.

Note that we could change the `SMALL_IMAGE_FACTOR` to 0.95, but then that would effectively mean that images smaller than 20 pixels will have a factor of 1.

That break down was just an educated guess to try to solve a problem I've seen. but we may as well change the thresholds and factors. Even add a MEDIUM image category.

I'm open to suggestions.

Reply:
I'm OK with this to start, we can always loosen if this ends up being overly aggressive.